tinyagentos/projects/task_store.py

```
@@ -389,10 +389,10 @@ async def update_task(
             # task was parked between that read and this write, the status edit
             # must not land (it was decided against a stale row).
             where += " AND status != 'parked'"
-        await self._db.execute(
-            f"UPDATE project_tasks SET {', '.join(sets)} WHERE {where}", params
-        )
-        await self._db.commit()
+        async with self._tx():
+            await self._db.execute(
+                f"UPDATE project_tasks SET {', '.join(sets)} WHERE {where}", params
```

Ruff (0.16.3)

[error] Possible SQL injection vector through string-based query construction

(S608)